### ISSUE TITLE
mariadb aborts and npm can not connect.

### ISSUE THREAD

**Issue by wschlotter** [jc21/nginx-proxy-manager#585]
I have my docker-compose file setup with the exact setup as the guide, and for the life of my i can not get the **app** to connect to the **db**

I am running OS X 10.15

i get the following output from the **db**
`2020-08-28T17:34:48.401631100Z [i] mysqld not found, creating....
2020-08-28T17:34:48.433280200Z [i] MySQL data directory not found, creating initial DBs
2020-08-28T17:34:49.731013700Z 2020-08-28 17:34:49 0 [ERROR] InnoDB: The Auto-extending innodb_system data file './ibdata1' is of a different size 0 pages than specified in the .cnf file: initial 768 pages, max 0 (relevant if non-zero) pages!
2020-08-28T17:34:49.734380300Z 2020-08-28 17:34:49 0 [ERROR] InnoDB: Plugin initialization aborted with error Generic error
2020-08-28T17:34:50.454179900Z 2020-08-28 17:34:50 0 [ERROR] Plugin 'InnoDB' init function returned error.
2020-08-28T17:34:50.454225800Z 2020-08-28 17:34:50 0 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.
2020-08-28T17:34:50.455927800Z 2020-08-28 17:34:50 0 [ERROR] Unknown/unsupported storage engine: InnoDB
2020-08-28T17:34:50.455966200Z 2020-08-28 17:34:50 0 [ERROR] Aborting
2020-08-28T17:34:50.486910400Z [i] Creating database: npm
2020-08-28T17:34:50.487263300Z [i] with character set: 'utf8' and collation: 'utf8_general_ci'
2020-08-28T17:34:50.487791800Z [i] Creating user: npm with password npm
2020-08-28T17:34:50.602434000Z 2020-08-28 17:34:50 0 [Note] /usr/bin/mysqld (mysqld 10.4.12-MariaDB) starting as process 55 ...
2020-08-28T17:34:50.752631200Z 2020-08-28 17:34:50 0 [Warning] Setting lower_case_table_names=2 because file system for /var/lib/mysql/ is case insensitive
2020-08-28T17:34:50.813506700Z 2020-08-28 17:34:50 0 [Note] InnoDB: Using Linux native AIO
2020-08-28T17:34:50.815307000Z 2020-08-28 17:34:50 0 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
2020-08-28T17:34:50.815598300Z 2020-08-28 17:34:50 0 [Note] InnoDB: Uses event mutexes
2020-08-28T17:34:50.815905700Z 2020-08-28 17:34:50 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
2020-08-28T17:34:50.816319300Z 2020-08-28 17:34:50 0 [Note] InnoDB: Number of pools: 1
2020-08-28T17:34:50.816635000Z 2020-08-28 17:34:50 0 [Note] InnoDB: Using SSE2 crc32 instructions
2020-08-28T17:34:50.817030900Z 2020-08-28 17:34:50 0 [Note] mysqld: O_TMPFILE is not supported on /var/tmp (disabling future attempts)
2020-08-28T17:34:50.844617300Z 2020-08-28 17:34:50 0 [Note] InnoDB: Initializing buffer pool, total size = 128M, instances = 1, chunk size = 128M
2020-08-28T17:34:50.871054000Z 2020-08-28 17:34:50 0 [Note] InnoDB: Completed initialization of buffer pool
2020-08-28T17:34:50.881261900Z 2020-08-28 17:34:50 0 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
2020-08-28T17:34:50.882717400Z 2020-08-28 17:34:50 0 [ERROR] InnoDB: The Auto-extending innodb_system data file './ibdata1' is of a different size 0 pages than specified in the .cnf file: initial 768 pages, max 0 (relevant if non-zero) pages!
2020-08-28T17:34:50.883440800Z 2020-08-28 17:34:50 0 [ERROR] InnoDB: Plugin initialization aborted with error Generic error
2020-08-28T17:34:51.389981400Z 2020-08-28 17:34:51 0 [Note] InnoDB: Starting shutdown...
2020-08-28T17:34:51.612766500Z 2020-08-28 17:34:51 0 [ERROR] Plugin 'InnoDB' init function returned error.
2020-08-28T17:34:51.613385100Z 2020-08-28 17:34:51 0 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.
2020-08-28T17:34:51.613625800Z 2020-08-28 17:34:51 0 [Note] Plugin 'FEEDBACK' is disabled.
2020-08-28T17:34:51.614058300Z 2020-08-28 17:34:51 0 [ERROR] Unknown/unsupported storage engine: InnoDB
2020-08-28T17:34:51.614222200Z 2020-08-28 17:34:51 0 [ERROR] Aborting
2020-08-28T17:34:51.634537800Z /scripts/run.sh: ignoring or entrypoint initdb empty /docker-entrypoint-initdb.d/*
2020-08-28T17:34:51.634863500Z
2020-08-28T17:34:51.634910300Z
2020-08-28T17:34:51.634931900Z MySQL init process done. Ready for start up.
2020-08-28T17:34:51.634952100Z
2020-08-28T17:34:51.635004100Z exec /usr/bin/mysqld --user=mysql --console --skip-name-resolve --skip-networking=0
2020-08-28T17:34:51.752410400Z 2020-08-28 17:34:51 0 [Note] /usr/bin/mysqld (mysqld 10.4.12-MariaDB) starting as process 1 ...
2020-08-28T17:34:51.802396600Z 2020-08-28 17:34:51 0 [Note] InnoDB: Using Linux native AIO
2020-08-28T17:34:51.804529400Z 2020-08-28 17:34:51 0 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
2020-08-28T17:34:51.805279500Z 2020-08-28 17:34:51 0 [Note] InnoDB: Uses event mutexes
2020-08-28T17:34:51.805473800Z 2020-08-28 17:34:51 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
2020-08-28T17:34:51.806061400Z 2020-08-28 17:34:51 0 [Note] InnoDB: Number of pools: 1
2020-08-28T17:34:51.806505900Z 2020-08-28 17:34:51 0 [Note] InnoDB: Using SSE2 crc32 instructions
2020-08-28T17:34:51.806951800Z 2020-08-28 17:34:51 0 [Note] mysqld: O_TMPFILE is not supported on /var/tmp (disabling future attempts)
2020-08-28T17:34:51.818032900Z 2020-08-28 17:34:51 0 [Note] InnoDB: Initializing buffer pool, total size = 128M, instances = 1, chunk size = 128M
2020-08-28T17:34:51.829776500Z 2020-08-28 17:34:51 0 [Note] InnoDB: Completed initialization of buffer pool
2020-08-28T17:34:51.845925200Z 2020-08-28 17:34:51 0 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
2020-08-28T17:34:51.847633800Z 2020-08-28 17:34:51 0 [ERROR] InnoDB: The Auto-extending innodb_system data file './ibdata1' is of a different size 0 pages than specified in the .cnf file: initial 768 pages, max 0 (relevant if non-zero) pages!
2020-08-28T17:34:51.848337900Z 2020-08-28 17:34:51 0 [ERROR] InnoDB: Plugin initialization aborted with error Generic error
2020-08-28T17:34:52.349825700Z 2020-08-28 17:34:52 0 [Note] InnoDB: Starting shutdown...
2020-08-28T17:34:52.572791500Z 2020-08-28 17:34:52 0 [ERROR] Plugin 'InnoDB' init function returned error.
2020-08-28T17:34:52.572885500Z 2020-08-28 17:34:52 0 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.
2020-08-28T17:34:52.573135900Z 2020-08-28 17:34:52 0 [Note] Plugin 'FEEDBACK' is disabled.
2020-08-28T17:34:52.592312900Z 2020-08-28 17:34:52 0 [ERROR] Could not open mysql.plugin table. Some plugins may be not loaded
2020-08-28T17:34:52.595071800Z 2020-08-28 17:34:52 0 [ERROR] Unknown/unsupported storage engine: InnoDB
2020-08-28T17:34:52.595127400Z 2020-08-28 17:34:52 0 [ERROR] Aborting`

and get the following error from the **app**

`2020-08-28T15:15:30.453259300Z finish: applet not found
2020-08-28T15:15:30.456431400Z s6-svscanctl: fatal: unable to control /var/run/s6/services: supervisor not listening
2020-08-28T15:15:30.459853400Z [cont-finish.d] executing container finish scripts...
2020-08-28T15:15:30.461984200Z [cont-finish.d] done.
2020-08-28T15:15:30.463155300Z [s6-finish] waiting for services.
2020-08-28T15:15:30.676793300Z [s6-finish] sending all processes the TERM signal.
2020-08-28T15:15:33.697539200Z [s6-finish] sending all processes the KILL signal and exiting.
2020-08-28T17:34:48.767249500Z [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
2020-08-28T17:34:48.847388300Z [s6-init] ensuring user provided files have correct perms...exited 0.
2020-08-28T17:34:48.849437800Z [fix-attrs.d] applying ownership & permissions fixes...
2020-08-28T17:34:48.853325700Z [fix-attrs.d] done.
2020-08-28T17:34:48.854915000Z [cont-init.d] executing container initialization scripts...
2020-08-28T17:34:48.857934000Z [cont-init.d] done.
2020-08-28T17:34:48.860285500Z [services.d] starting services
2020-08-28T17:34:48.941085800Z [services.d] done.
2020-08-28T17:34:49.165439900Z ❯ Enabling IPV6 in hosts: /etc/nginx/conf.d
2020-08-28T17:34:49.165566000Z   ❯ /etc/nginx/conf.d/production.conf
2020-08-28T17:34:49.181385700Z   ❯ /etc/nginx/conf.d/include/letsencrypt-acme-challenge.conf
2020-08-28T17:34:49.200511100Z   ❯ /etc/nginx/conf.d/include/assets.conf
2020-08-28T17:34:49.208384700Z   ❯ /etc/nginx/conf.d/include/ssl-ciphers.conf
2020-08-28T17:34:49.215368900Z   ❯ /etc/nginx/conf.d/include/block-exploits.conf
2020-08-28T17:34:49.232412600Z   ❯ /etc/nginx/conf.d/include/proxy.conf
2020-08-28T17:34:49.243586100Z   ❯ /etc/nginx/conf.d/include/force-ssl.conf
2020-08-28T17:34:49.252971600Z   ❯ /etc/nginx/conf.d/include/ip_ranges.conf
2020-08-28T17:34:49.280353900Z   ❯ /etc/nginx/conf.d/include/resolvers.conf
2020-08-28T17:34:49.289197100Z   ❯ /etc/nginx/conf.d/default.conf
2020-08-28T17:34:49.587980600Z ❯ Enabling IPV6 in hosts: /data/nginx
2020-08-28T17:34:52.918749400Z [8/28/2020] [5:34:52 PM] [Global   ] › ✖  error     connect ECONNREFUSED 172.18.0.3:3306
2020-08-28T17:34:53.934571000Z [8/28/2020] [5:34:53 PM] [Global   ] › ✖  error     getaddrinfo ENOTFOUND db`

Here is my docker-compose.yml file setup:

`version: '3'
services:
  app:
    image: 'jc21/nginx-proxy-manager:latest'
    container_name: app
    ports:
      - '80:80'
      - '81:81'
      - '443:443'
    volumes:
      - ./config.json:/app/config/production.json
      - ./data:/data
      - ./letsencrypt:/etc/letsencrypt

  db:
    image: 'jc21/mariadb-aria:10.4'
    container_name: db
    environment:
      MYSQL_ROOT_PASSWORD: 'npm'
      MYSQL_DATABASE: 'npm'
      MYSQL_USER: 'npm'
      MYSQL_PASSWORD: 'npm'
    volumes:
      - ./data/mysql:/var/lib/mysql`

here's my config.json file:

`{
    "database": {
        "engine": "mysql",
        "host": "db",
        "name": "npm",
        "user": "npm",
        "password": "npm",
        "port": 3306
    }
}`

**Comment by wschlotter** [comment]
After doing some digging, i am coming to the conclusion that the user is not being set withing the database.

**Comment by codeagencybe** [comment]
Is there any update yet about this? 
This bug is mentioned dozens of time and seems like there is no fix coming??

I have exactly the same problem. 
Everything is identical from the setup examples and it's not working. 

`db_1 | 2020-09-19 12:31:16 0 [ERROR] InnoDB: The Auto-extending innodb_system data file './ibdata1' is of a different size 0 pages than specified in the .cnf file: initial 768 pages, max 0 (relevant if non-zero) pages!

db_1 | 2020-09-19 12:31:16 0 [ERROR] InnoDB: Plugin initialization aborted with error Generic error

db_1 | 2020-09-19 12:31:17 0 [ERROR] Plugin 'InnoDB' init function returned error.

db_1 | 2020-09-19 12:31:17 0 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.

db_1 | 2020-09-19 12:31:17 0 [ERROR] Unknown/unsupported storage engine: InnoDB
`

**Comment by codeagencybe** [comment]
@wschlotter 

I got it working!

``db_1 | 2020-09-19 12:31:16 0 [ERROR] InnoDB: The Auto-extending innodb_system data file './ibdata1' is of a different size 0 pages than specified in the .cnf file: initial 768 pages, max 0 (relevant if non-zero) pages!

`db_1 | 2020-09-19 12:31:16 0 [ERROR] InnoDB: The Auto-extending innodb_system data file './ibdata1' is of a different size 0 pages than specified in the .cnf file: initial 768 pages, max 0 (relevant if non-zero) pages!`



All I did was delete the .ibdata1 file that shows 0kb
And then re-deploy the whole setup. Now it generates a new clean .ibdata file that is around 12Mb size
